rtrlib/aspa/aspa_array/aspa_array.c

```diff
@@ -156,7 +156,7 @@ enum aspa_status aspa_array_remove(struct aspa_array *array, size_t index)
 
 inline struct aspa_record *aspa_array_get_record(struct aspa_array *array, size_t index)
 {
-	if (!array || index >= array->size || array->size == 0 || !array->data)
+	if (!array || index >= array->size || !array->data) // or array->size == 0
```

Comment:
why removing this extra check ?

Reply:
because `size_t` is unsigned, `array_size == 0` will always be caught by `index >= array->size`

Reply:
I don't think that removing the extra check has any kind of performance impact because the if branch is entered right after you encounter the first `true` condition